shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -921,6 +927,42 @@ - (void)waitForFirstFrame:(NSTimeInterval)timeout
   });
 }
 
+- (FlutterEngine*)spawnWithEntrypoint:(NSString*)entrypoint {
+  assert(_shell);
+  FlutterEngine* result =
+      [[FlutterEngine alloc] initWithName:[_labelPrefix stringByAppendingString:@"-spawn"]
+                                  project:_dartProject.get()
+                   allowHeadlessExecution:_allowHeadlessExecution];
+
+  flutter::Settings settings = _shell->GetSettings();
+  if (entrypoint) {
+    settings.advisory_script_entrypoint = entrypoint.UTF8String;
+    settings.advisory_script_uri = std::string("main.dart");
+  } else {
+    settings.advisory_script_entrypoint = std::string("main");
+    settings.advisory_script_uri = std::string("main.dart");
+  }
+
+  flutter::Shell::CreateCallback<flutter::PlatformView> on_create_platform_view =
+      [self](flutter::Shell& shell) {
+        [self recreatePlatformViewController];
+        return std::make_unique<flutter::PlatformViewIOS>(
+            shell, self->_renderingApi, self->_platformViewsController, shell.GetTaskRunners());
+      };
+
+  flutter::Shell::CreateCallback<flutter::Rasterizer> on_create_rasterizer =
+      [](flutter::Shell& shell) { return std::make_unique<flutter::Rasterizer>(shell); };
+
+  std::unique_ptr<flutter::Shell> shell =
+      _shell->Spawn(std::move(settings), on_create_platform_view, on_create_rasterizer);
+
+  result->_threadHost = _threadHost;
```

Comment:
Oh, so this fixes half of https://github.com/flutter/flutter/issues/72019 too?

Reply:
Yep, I realized that after the fact.  Forgot I did thread sharing in first pr.